src/Microsoft.ComponentDetection.Detectors/pnpm/Pnpm6ComponentDetector.cs

```diff
@@ -0,0 +1,159 @@
+namespace Microsoft.ComponentDetection.Detectors.Pnpm;
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.ComponentDetection.Contracts;
+using Microsoft.ComponentDetection.Contracts.Internal;
+using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using Microsoft.Extensions.Logging;
+
+public class Pnpm6ComponentDetector : FileComponentDetector, IExperimentalDetector
+{
+    public Pnpm6ComponentDetector(
+        IComponentStreamEnumerableFactory componentStreamEnumerableFactory,
+        IObservableDirectoryWalkerFactory walkerFactory,
+        ILogger<Pnpm6ComponentDetector> logger)
+    {
+        this.ComponentStreamEnumerableFactory = componentStreamEnumerableFactory;
+        this.Scanner = walkerFactory;
+        this.NeedsAutomaticRootDependencyCalculation = true;
```

Comment:
nit: move this out of the constructor `public override bool NeedsAutomaticRootDependencyCalculation => true;`